### DISCUSSION TITLE
Study material on how to use & apply AdvantEDGE in use cases

### DISCUSSION THREAD

**Discussion by anokun7** [InterDigitalInc/AdvantEDGE#216] [General]
So I have been able to setup AdvantEdge on kubernetes and also played with the example demos. I intend to give my team an overview of AdvantEdge and how it can be leveraged. But I could not find much material on the internet.

Is there some material already available for how to generally use the tool? Not talking about building or setting it up, but once setup, how to actually have an application - maybe as a slide deck or a document etc. It would be great if it would showcase specific use cases.

Thank you very much.

**Comment by roymx** [comment]
Hello @anokun7 ,

Thank you for reaching out, it's a good question.

The platform can be used in different ways depending on your own objectives; allow me some time to gather exemplary use cases that I'll share soon.

Then, we can then have a discussion if you have more questions.

**Reply by anokun7** [reply]
Thank you. I look forward to the material.

**Reply by roymx** [reply]
As promised, here are resources & sorry for the long answer.

We will first introduce use cases then focus on usage workflow.

As a reminder:
The platform focuses on providing researcher & application developers the ability to conduct edge computing experiments and research.
It does so in two ways: 
 1) edge network emulation: providing configurability of latency/throughput/packet loss in a controlled environment
 2) MEC Services APIs: providing edge-specific APIs to push the edge-native application concepts 
 
Other aspects of the platform facilitate scenarios implementation by providing GIS service, databases, visualizations, etc.

Use cases:
- Cloud service to demonstrate ETSI MEC APIs
  - Their focus is on the MEC Service aspect.
  - https://try-mec.etsi.org/ (you can try it...)
  - Provides users with a Sandbox environment to experiment & learn about ETSI MEC APIs
  - Currently supports Location Service, Radio Network Interface Service (RNIS), WLAN Access Information (WAI)
  - New services & scenarios will be added in 2021
  
- Research tool in the Living Edge Lab at CMU University in Pittsburgh
  - Their focus is on Edge Network Emulation aspects
  - "Noisy Neighbors" – Edge Multi-Tenancy Research (https://youtu.be/uDkaXd_yWIk)
  - "A Virtual Walk down Walnut Street"(https://youtu.be/4Hb1ibgV-AM)
  - Research paper - http://reports-archive.adm.cs.cmu.edu/anon/2020/CMU-CS-20-135.pdf

- Research tool in FH Dortmund
  - Their focus is on the Edge Network Emulation aspects
  - Think they have an on-prem deployment so students can use it for labs
  - Research paper - https://ieeexplore.ieee.org/document/9297078

- We also use it to conduct our internal experiments
  - We did test several edge use-cases with it.
  - To name a few: interactive gaming, live video stylization, mission critical automation with drones & UTM, mixed reality

Getting started:
We do not have a great step-by-step document as you pointed out - other than the demo1 example.
We will add this on the roadmap.

Generally, users of the platform:
  - have an edge computing concept/idea that they want to validate
  - do not have access to an edge computing environment to test it
  - or have access to an edge computing environment where they cannot control network topology or characteristics
  
Pre-requisites:
- A containerized client application (alternatively, client app can run on an external device such as a phone)
- A containerized server application (alternatively, it can run externally too -- but this is a less common use case)

We will call these applications the "vertical".

To get started running your vertical on AdvantEDGE:
(1) Configuration
- Build your scenario - topology (cloud, edge nodes, Point of Access, etc.)
- Use geolocation to place PoAs which in turn will allow some automation (more on this later)
- Position Terminal(s), Edge Nodes and Processes (edge/cloud applications, etc.) in the network topology
- Save your scenario

(2) Deployment
- Deploy your scenario
- Once deployed, the platform logs network information 
  - In most cases, it is useful to instrument the vertical to log metrics which will allows to visualize both network & vertical metrics together.
- Manual event generation such as Mobility, Network Characteristics Change, etc. allow to visualize the effect on the application
  - When moving the terminal, network characteristics change in real time
  - Changing network characteristics, also updates in real time
- If PoAs are geolocated – this enables automations
  - Network characteristics can vary according to the position of the terminal to the PoA
  - Mobility automation can connect the terminal to the closest PoA as the terminal move
  - Terminal can have path & speed configured and move autonomously
  - etc.
- Events are recorded as you deploy the scenario and can therefore be replayed / saved for later re-use
  - In an experimentation setting, this allows to repeat experiments varying the workload

(3) Visualize your data
- No matter the experiment, metrics and visibility are key
- The platform offers a time series database (InfluxDB) and a visualization tool (Grafana) to allows aggregating metrics and visualizing experiment data.

I think this summarizes the platform usage workflow; as you can see, it can be used for multiple purposes.

Finally, we open-sourced this tool to foster collaboration on edge-computing.
We are open to setup a call if you have interest.

**Reply by anokun7** [reply]
This is simply too awesome! I can't thank you enough for these pointers and for taking the time to put this together. I plan to go through all of these this weekend.

**Reply by roymx** [reply]
FYI if you are interested - related to the references provided above.

Next week is the Edge Computing World Europe virtual conference and I believe it's free to attend.
https://www.edgecomputingworld.com/

Some folks from CMU will be presenting: Prof. Satya and Jim Blakely (director of LEL) as well as some folks from ETSI: Alex Reznik (ETSI MEC Chair)

Not sure if they will be talking about AdvantEDGE - but anyhow still a free conference on Edge Computing.

**Reply by anokun7** [reply]
Great. I have signed up & will be in attendance at the conference.
